pkg/test/case.go

```diff
@@ -397,9 +402,17 @@ func (t *Case) determineNamespace() *namespace {
 	if t.PreferredNamespace == "" {
 		ns.Name = fmt.Sprintf("kuttl-test-%s", petname.Generate(2, "-"))
 		ns.AutoCreated = true
+	} else {
+		exist, err := t.NamespaceExists(t.PreferredNamespace)
+		if err != nil {
+			return nil, fmt.Errorf("found namespace %s err, %s", t.PreferredNamespace, err.Error())
```

Comment:
I think something like this would be more fitting and robust to weird ns names:
```suggestion
			return nil, fmt.Errorf("failed to determine existence of namespace %q: %w", t.PreferredNamespace, err)
```